File: minimum-depth-of-binary-tree/minimum_depth_of_binary_tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func minDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    left := minDepth(root.Left)
    right := minDepth(root.Right)
    if left == 0 || right == 0 {
        return left + right + 1
    } else if left >= right {
        return right + 1
    } else {
        return left + 1
    }
}

File: minimum-depth-of-binary-tree/minimum_depth_of_binary_tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (!root) {
        return 0;
    }
    var left = minDepth(root.left)
    var right = minDepth(root.right)
    if (!left || !right) {
        return left + right + 1;
    } else if (left >= right) {
        return right + 1;
    } else {
        return left + 1;
    }
};